pkg/service/controller/servingcert/secret_updating_controller.go

```diff
@@ -234,7 +236,13 @@ func (sc *ServiceServingCertUpdateController) syncSecret(key string) error {
 	dnsName := secret.Annotations[ServiceNameAnnotation] + "." + secret.Namespace + ".svc"
 	fqDNSName := dnsName + "." + sc.dnsSuffix
 	certificateLifetime := 365 * 2 // 2 years
-	servingCert, err := sc.ca.MakeServerCert(sets.NewString(dnsName, fqDNSName), certificateLifetime)
+	servingCert, err := sc.ca.MakeServerCert(
+		sets.NewString(dnsName, fqDNSName),
+		certificateLifetime,
+		extensions.ServiceServerCertificateExtension(&kapi.Service{
+			ObjectMeta: kapi.ObjectMeta{UID: types.UID(secret.Annotations[ServiceUIDAnnotation])},
```

Comment:
yeah, which doesn't seem great. I expected at least verification of the data on the service object itself